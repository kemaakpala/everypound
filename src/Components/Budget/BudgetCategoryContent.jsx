import Grid from '@material-ui/core/Grid';

const BudgetCategoryContent = ({children}) => {
  return (
    <Grid
      item xs={12} 
      // style={{ border: '4px solid green' }}
    >
      {children}
    </Grid>
  )
}

export default BudgetCategoryContent;