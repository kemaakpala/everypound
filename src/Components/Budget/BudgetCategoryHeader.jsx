import Grid from '@material-ui/core/Grid';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
const useStyles = makeStyles((theme) => ({
  categoryLabel: {
    display: 'flex',
    flex: 1,
    alignItems: 'center',
    '& h1': {
      paddingLeft: theme.spacing(1)
    }
  }
}));
const BudgetCategoryHeader = ({ title }) => {
  const classes = useStyles();
  return (
    <Grid item xs={12}
    // style={{ border: '1px solid red' }}
    >
      <Grid container spacing={1} alignItems="center"
      // style={{ border: '1px solid blue' }}
      >
        <Grid item xs={4}>
          <div className={classes.categoryLabel}>
            <Avatar>{title[0].toUpperCase()}</Avatar>
            <Typography variant="h6" component='h1'>{title}</Typography>
          </div>
        </Grid>
        <Grid item xs={3}><Typography variant="h6" component='h1' align="right">Planned</Typography></Grid>
        <Grid item xs={3}><Typography variant="h6" component='h1' align="right">Actual</Typography></Grid>
        <Grid item xs={2}><Typography variant="h6" component='h1' align="right"></Typography></Grid>
      </Grid>
    </Grid>
  )
}

export default BudgetCategoryHeader;