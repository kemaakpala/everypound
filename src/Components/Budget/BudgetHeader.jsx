import Grid from '@material-ui/core/Grid';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
const useStyles = makeStyles((theme) => ({
  groupLabel: {
    display: 'flex',
    flex: 1,
    alignItems: 'center',
    '& p': {
      color: 'red',
      paddingLeft: theme.spacing(1)
    }
  }
}));
const BudgetHeader = ({ groupTitle }) => {
  const classes = useStyles();
  return (
    <Grid item xs={12} style={{ border: '1px solid red' }}>
      <Grid container spacing={1} alignItems="center" style={{ border: '1px solid blue' }}>
        <Grid item xs={4}>
          <div className={classes.groupLabel}>
            <Avatar>{groupTitle[0].toUpperCase()}</Avatar>
            <Typography>{groupTitle}</Typography>
          </div>
        </Grid>
        <Grid item xs={4}><Typography>Planned</Typography></Grid>
        <Grid item xs={4}><Typography>Actual</Typography></Grid>
      </Grid>
    </Grid>
  )
}

export default BudgetHeader;