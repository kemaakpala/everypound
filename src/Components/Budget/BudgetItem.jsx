import {useState} from 'react';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import InputAdornment from '@material-ui/core/InputAdornment';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import { makeStyles } from '@material-ui/core/styles';
import {parseFloatAmount} from '../../utils/numberUtils';


const useStyles = makeStyles((theme) => ({
  budgetItemContainer: {
    paddingTop: theme.spacing(1),
    paddingBottom: theme.spacing(1)
  },
  deleteBtn: {
    marginLeft: theme.spacing(1),
    padding: 0
  },
  planned: {
    '& input': {
      textAlign: 'right'
    }
  }
}));
const BudgetItem = ({name}) => {
  const classes = useStyles();
  const [planned, setPlanned] = useState(parseFloat('0').toFixed(2));

  const handleChange = (e) => { setPlanned(e.target.value) }
  const handleBlur = (e) => {
    const val = parseFloatAmount(e.target.value)
    const isNan = window.isNaN(val);
    setPlanned(parseFloatAmount('0'));
    if (!isNan){
      setPlanned(val)
    }
  }

  return (
    <>
      <Grid className={classes.budgetItemContainer} container spacing={1} alignItems="flex-end">
        <Grid item xs={4}>
          <Typography>{name}</Typography>
        </Grid>
        <Grid item xs={3} align="right">
          <TextField
            className={classes.planned}
            InputProps={{
              startAdornment: <InputAdornment position="start">£</InputAdornment>
            }}
            value={planned}
            onChange={handleChange}
            onBlur={handleBlur}
          />
        </Grid>
        <Grid item xs={3} align="right">
          <Typography>£0.00</Typography>
        </Grid>
        <Grid item xs={2} align="right">
        <IconButton aria-label="delete" className={classes.deleteBtn}>
          <DeleteIcon fontSize="small" />
        </IconButton>
        </Grid>
      </Grid>
    </>
  )
}

export default BudgetItem;