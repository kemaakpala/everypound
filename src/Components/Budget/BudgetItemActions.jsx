import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Divider from '@material-ui/core/Divider';

const BudgetItemActions = () => (
  <Grid item xs={12}>
    <Divider />
    <Button>Add Item</Button>
  </Grid>
)

export default BudgetItemActions;