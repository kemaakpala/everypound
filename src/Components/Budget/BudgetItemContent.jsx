import Grid from '@material-ui/core/Grid';

const BudgetItemContainer = ({children}) => {
  return (
    <Grid
      item xs={12} 
      // style={{ border: '4px solid green' }}
    >
      {children}
    </Grid>
  )
}

export default BudgetItemContainer;