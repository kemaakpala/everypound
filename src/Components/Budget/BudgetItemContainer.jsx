import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import Divider from '@material-ui/core/Divider';
import { makeStyles } from '@material-ui/core/styles';
import BudgetItemActions from './BudgetItemActions';
const useStyles = makeStyles((theme) => ({
  root: {
    marginBottom: theme.spacing(1),
    padding: theme.spacing(1)
  }
}))
const BudgetItemContainer = ({ children }) => {
  const classes = useStyles();
  return (
    <Paper className={classes.root} elevation={0} variant="outlined">
      <Grid container spacing={2}
      // style={{ border: '1px solid black', margin: '8px 0 0 0' }}
      >
        <Grid item xs={12}
        // style={{ border: '4px solid green' }}
        > {children}</Grid>
        <BudgetItemActions />
      </Grid>
    </Paper>
  )
}

export default BudgetItemContainer;