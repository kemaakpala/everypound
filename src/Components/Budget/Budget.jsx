import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Divider from '@material-ui/core/Divider';
import BudgetHeader from './BudgetHeader';
import BudgetItemContainer from './BudgetItemContainer';
import BudgetItemContent from './BudgetItemContent';
import BudgetItem from './BudgetItem';

const Budget = () => {
  return (
    <Grid item xs={12} sm={8} md={8}>
      <BudgetItemContainer>
        <BudgetHeader groupTitle="Housing" />
        <BudgetItemContent>
          <BudgetItem name="Mortgage / Rent" />
          <Divider />
          <BudgetItem name="Water Bi annual Bill" />
          <Divider />
          <BudgetItem name="Natural Gas" />
          <Divider />
          <BudgetItem name="Electricity" />
        </BudgetItemContent>
      </BudgetItemContainer>

      <BudgetItemContainer>
        <BudgetHeader groupTitle="Food" />
        <BudgetItemContent>
          <BudgetItem name="Groceries" />
          <Divider />
          <BudgetItem name="Restaurant" />
          <Divider />
          <BudgetItem name="Lunch" />
          <Divider />
          <BudgetItem name="Take out" />
        </BudgetItemContent>
      </BudgetItemContainer>

      <BudgetItemContainer>
        <BudgetHeader groupTitle="Personal" />
        <BudgetItemContent>
          <BudgetItem name="Phone Bill" />
          <Divider />
          <BudgetItem name="Wifeys Phone Bill" />
          <Divider />
          <BudgetItem name="Allowance" />
          <Divider />
          <BudgetItem name="Wifeys Allowance" />
        </BudgetItemContent>
      </BudgetItemContainer>
    </Grid>
  )
}

export default Budget;