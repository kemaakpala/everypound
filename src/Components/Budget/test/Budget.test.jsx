import Budget from '../Budget';
import { render } from '@testing-library/react';

describe('<Budget/>', () => {
  it('matches snapshot', () => {
    const { firstChild } = render(<div><Budget /></div>)
    expect(firstChild).toMatchSnapshot();
  })
})