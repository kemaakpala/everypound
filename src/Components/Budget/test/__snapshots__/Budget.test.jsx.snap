// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Budget/> matches snapshot 1`] = `undefined`;
